On the first test run, there is only one real failure (mypy). The second failure at
```
 ======================================================================
FAIL: test_session_attendance (ietf.meeting.tests_views.ProceedingsTests.test_session_attendance)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/__w/datatracker/datatracker/ietf/meeting/tests_views.py", line 9323, in test_session_attendance
    self.assertContains(r, person.plain_name())
  File "/usr/local/lib/python3.9/site-packages/django/test/testcases.py", line 660, in assertContains
    self.assertTrue(
AssertionError: False is not true : Couldn't find 'David O'Connor' in response
```
is a transient.
The rest are just consequences of the first two.

For the first error, try 
```
HistoricalReviewAssignment = ReviewAssignment.history.model
```
just before using it in the code rather than importing HistoricalReviewAssignment.

--- a/ietf/group/tests_review.py
+++ b/ietf/group/tests_review.py
@@ -815,3 +815,131 @@ class ResetNextReviewerInTeamTests(TestCase):
             self.assertEqual(NextReviewerInTeam.objects.get(team=group).next_reviewer, reviewers[target_index].person)
             self.client.logout()
             target_index += 2
+
+class RequestsHistoryTests(TestCase):
+    def test_requests_history_overview_page(self):
+        # Make assigned assignment
+        review_req = ReviewRequestFactory(state_id='assigned')
+        assignment = ReviewAssignmentFactory(review_request=review_req,
+                                             state_id='assigned',
+                                             reviewer=EmailFactory(),
+                                             assigned_on = review_req.time)
+        group = review_req.team
+
+        for url in [urlreverse(ietf.group.views.review_requests_history,
+                               kwargs={ 'acronym': group.acronym }),
+                    urlreverse(ietf.group.views.review_requests_history,
+                               kwargs={ 'acronym': group.acronym ,
+                                        'group_type': group.type_id}),
+                    urlreverse(ietf.group.views.review_requests_history,
+                               kwargs={ 'acronym': group.acronym }) +
+                    '?since=3m',
+                    urlreverse(ietf.group.views.review_requests_history,
+                               kwargs={ 'acronym': group.acronym ,
+                                        'group_type': group.type_id }) +
+                    '?since=3m']:
+            r = self.client.get(url)
+            self.assertEqual(r.status_code, 200)
+            self.assertContains(r, review_req.doc.name)
+            self.assertContains(r, 'Assigned')
+            self.assertContains(r, escape(assignment.reviewer.person.name))
+
+        url = urlreverse(ietf.group.views.review_requests_history,
+                         kwargs={ 'acronym': group.acronym })
+
+        assignment.state = ReviewAssignmentStateName.objects.get(slug="completed")
+        assignment.result = ReviewResultName.objects.get(slug="ready")
+        assignment.save()
+
+        r = self.client.get(url)
+        self.assertEqual(r.status_code, 200)
+        self.assertContains(r, review_req.doc.name)
+        self.assertContains(r, 'Assigned')
+        self.assertContains(r, 'Completed')
+
+    def test_requests_history_filter_page(self):
+        # First assignment as assigned
+        review_req = ReviewRequestFactory(state_id = 'assigned',
+                                          doc = DocumentFactory())
+        assignment = ReviewAssignmentFactory(review_request = review_req,
+                                             state_id = 'assigned',
+                                             reviewer = EmailFactory(),
+                                             assigned_on = review_req.time)
+        group = review_req.team
+
+        # Second assignment in same group as accepted
+        review_req2 = ReviewRequestFactory(state_id = 'assigned',
+                                           team = review_req.team,
+                                           doc = DocumentFactory())
+        assignment2 = ReviewAssignmentFactory(review_request = review_req2,
+                                              state_id='accepted',
+                                              reviewer = EmailFactory(),
+                                              assigned_on = review_req2.time)
+
+        # Modify the assignment to be completed, and mark it ready
+        assignment2.state = ReviewAssignmentStateName.objects.get(slug="completed")
+        assignment2.result = ReviewResultName.objects.get(slug="ready")
+        assignment2.save()
+
+        # Check that we have all information when we do not filter
+        url = urlreverse(ietf.group.views.review_requests_history,
+                         kwargs={ 'acronym': group.acronym })
+        r = self.client.get(url)
+        self.assertEqual(r.status_code, 200)
+        self.assertContains(r, review_req.doc.name)
+        self.assertContains(r, review_req2.doc.name)
+        self.assertContains(r, 'Assigned')
+        self.assertContains(r, 'Accepted')
+        self.assertContains(r, 'Completed')
+        self.assertContains(r, 'Ready')
+        self.assertContains(r, escape(assignment.reviewer.person.name))
+        self.assertContains(r, escape(assignment2.reviewer.person.name))
+
+        # Check first reviewer history
+        for url in [urlreverse(ietf.group.views.review_requests_history,
+                               kwargs={ 'acronym': group.acronym }) +
+                    '?reviewer_email=' + str(assignment.reviewer),
+                    urlreverse(ietf.group.views.review_requests_history,
+                               kwargs={ 'acronym': group.acronym ,
+                                        'group_type': group.type_id}) +
+                    '?reviewer_email=' + str(assignment.reviewer)]:
+            r = self.client.get(url)
+            self.assertEqual(r.status_code, 200)
+            self.assertContains(r, review_req.doc.name)
+            self.assertNotContains(r, review_req2.doc.name)
+            self.assertContains(r, 'Assigned')
+            self.assertNotContains(r, 'Accepted')
+            self.assertNotContains(r, 'Completed')
+            self.assertNotContains(r, 'Ready')
+            self.assertContains(r, escape(assignment.reviewer.person.name))
+            self.assertNotContains(r, escape(assignment2.reviewer.person.name))
+
+        # Check second reviewer history
+        for url in [urlreverse(ietf.group.views.review_requests_history,
+                               kwargs={ 'acronym': group.acronym }) +
+                    '?reviewer_email=' + str(assignment2.reviewer),
+                    urlreverse(ietf.group.views.review_requests_history,
+                               kwargs={ 'acronym': group.acronym ,
+                                        'group_type': group.type_id}) +
+                    '?reviewer_email=' + str(assignment2.reviewer)]:
+            r = self.client.get(url)
+            self.assertEqual(r.status_code, 200)
+            self.assertNotContains(r, review_req.doc.name)
+            self.assertContains(r, review_req2.doc.name)
+            self.assertNotContains(r, 'Assigned')
+            self.assertContains(r, 'Accepted')
+            self.assertContains(r, 'Completed')
+            self.assertContains(r, 'Ready')
+            self.assertNotContains(r, escape(assignment.reviewer.person.name))
+            self.assertContains(r, escape(assignment2.reviewer.person.name))
+
+        # Check for reviewer that does not have anything
+        url = urlreverse(ietf.group.views.review_requests_history,
+                         kwargs={ 'acronym': group.acronym }) + '?reviewer_email=nobody@nowhere.example.org'
+
+        r = self.client.get(url)
+        self.assertEqual(r.status_code, 200)
+        self.assertNotContains(r, review_req.doc.name)
+        self.assertNotContains(r, 'Assigned')
+        self.assertNotContains(r, 'Accepted')
+        self.assertNotContains(r, 'Completed')

--- a/ietf/group/views.py
+++ b/ietf/group/views.py
@@ -92,7 +92,7 @@ from ietf.meeting.utils import group_sessions
 from ietf.name.models import GroupTypeName, StreamName
 from ietf.person.models import Email, Person
 from ietf.review.models import (ReviewRequest, ReviewAssignment, ReviewerSettings, 
-                                ReviewSecretarySettings, UnavailablePeriod, HistoricalReviewAssignment )
+                                ReviewSecretarySettings, UnavailablePeriod )
 from ietf.review.policies import get_reviewer_queue_policy
 from ietf.review.utils import (can_manage_review_requests_for_team,
                                can_access_review_stats_for_team,
@@ -692,13 +692,13 @@ def review_requests_history(request, acronym, group_type=None):
         raise Http404
 
     reviewer_email = request.GET.get("reviewer_email", None)
-    
+
     if reviewer_email:
-        history = HistoricalReviewAssignment.objects.filter(
+        history = ReviewAssignment.history.model.objects.filter(
             review_request__team__acronym=acronym,
             reviewer=reviewer_email)
     else:
-        history = HistoricalReviewAssignment.objects.filter(
+        history = ReviewAssignment.history.model.objects.filter(
             review_request__team__acronym=acronym)
         reviewer_email = ''
         

--- a/ietf/templates/group/review_requests_history.html
+++ b/ietf/templates/group/review_requests_history.html
@@ -16,7 +16,8 @@
     {% endif %}
     <form class="reviews-history-filter" action="#reviewshistory">
       <div>
-	Reviewer filter:
+	<label for="reviewer_email"
+	       class="query-input-label">Reviewer filter:</label>
 	<input type="text" class="query-input" id="reviewer_email"
 	       value="{{ reviewer_email }}" name="reviewer_email">
 	<button class="btn btn-primary" type="submit">Filter</button>
@@ -42,11 +43,6 @@
                 <th scope="col" data-sort="by">By</th>
                 <th scope="col" data-sort="doc">Document</th>
                 <th scope="col" data-sort="state">State</th>
-		<!-- 
-                     <th scope="col" data-sort="assigned">Assigned</th>
-                     <th scope="col"
-                     data-sort="completed">Completed</th>
-		-->
                 <th scope="col" data-sort="review">Reviewer</th>
                 <th scope="col" data-sort="result">Result</th>
                 <th scope="col" data-sort="description">Description</th>
@@ -65,23 +61,23 @@
 			  {% else %}
 			  <a href="{% url "ietf.doc.views_doc.document_main" name=h.review_request.doc.name %}">{{ h.review_request.doc.name }}</a>
 			  {% endif %}
-                        <td><!-- {{ h.state }} for sort-->
+			</td>
+                        <td data-text="{{ h.state }}">
 			  <a href="{% url "ietf.doc.views_review.review_request" name=h.review_request.doc.name request_id=h.review_request.pk %}">{{ h.state }} </a>
-			<!--   
-                             <td>{{ h.assigned_on|date }} 
-                             <td>{{ h.completed_on|date }}
-			-->
-                        <td> <!-- {{ h.reviewer }} for sort -->
+			</td>
+                        <td data-text="{{ h.reviewer }}">
 			  {% person_link h.reviewer.person %}
 			  <a href="?reviewer_email={{h.reviewer.email_address}}#reviewshistory">
 			    (set&nbsp;as&nbsp;filter)
 			  </a>
-                        <td><!-- h.result for sort -->
+			</td>
+                        <td data-text="{{ h.result }}">
 			  {% if h.review %}<a href="{{ h.review.get_absolute_url }}">
 			    {{ h.result }}</a>
 			  {% else %}
 			  {{ h.result }}
 			  {% endif %}
+			</td>
                         <td>{{ h.history_change_reason }}</td>
                     </tr>
                 {% endfor %}

